background/main.ts

```diff
@@ -965,9 +965,9 @@ export default class Main extends BaseService<never> {
           const signedTransactionResult =
             await this.signingService.signTransaction(request, accountSigner)
           await this.store.dispatch(transactionSigned(signedTransactionResult))
-          transactionConstructionSliceEmitter.emit("signedTransactionResult", 
+          setTimeout(() => transactionConstructionSliceEmitter.emit("signedTransactionResult", 
             signedTransactionResult
-          )
+          ), 1000) // could check broadcastOnSign here and broadcast if false but this is a hacky solution (could result in tx broadcasted twice)
```

Comment:
So IIUC this will not retry, correct? If broadcast fails, is there a way for a user to manually rebroadcast?

Reply:
In some cases it broadcasts the same transaction twice. This does not cause any issue to the user (and the user is not aware) but it is a bug that should be fixed. https://github.com/PelagusWallet/pelagus-extension/issues/37
If broadcasting a transaction fails for some reason, the user will need to manually attempt to broadcast the transaction again.